Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/mesh_smoothing.h

```diff
@@ -0,0 +1,346 @@
+
+  remesher.init_smoothing(faces);
+
+  #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+  t.stop();
+  std::cout << " done ("<< t.time() <<" sec)." << std::endl;
+  std::cout << "#iter = " << nb_iterations << std::endl;
+  std::cout << "Smoothing ..." << std::endl;
+  t.reset(); t.start();
+  #endif
+
+  for(std::size_t i=0; i<nb_iterations; ++i)
+  {
+    #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+    std::cout << " * Iteration " << (i + 1) << " *" << std::endl;
+    #endif
+
+    remesher.angle_relaxation();
+    if(do_project)
+      remesher.project_to_surface();
+  }
+
+  #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+  t.stop();
+  std::cout << "Smoothing done in ";
+  std::cout << t.time() << " sec." << std::endl;
+  #endif
+
+}
+
+template<typename PolygonMesh, typename NamedParameters>
+void smooth_angles(PolygonMesh& pmesh, const NamedParameters& np)
+{
+  smooth_angles(faces(pmesh), pmesh, np);
+}
+
+template<typename PolygonMesh>
+void smooth_angles(PolygonMesh& pmesh)
+{
+  smooth_angles(faces(pmesh), pmesh, parameters::all_default());
+}
+
+/*!
+* \ingroup PMP_meshing_grp
+* smoothes a triangulated region of a polygon mesh using area based criteria.
+* This function tries to make the triangle area distribution as uniform as possible
+* by moving non constrained vertices.
+* Optionally, the points are reprojected after each iteration.
+*
+* @tparam PolygonMesh model of `MutableFaceGraph`.
+*         If `PolygonMesh` has an internal property map for `CGAL::face_index_t`,
+*         and no `face_index_map` is given
+*         as a named parameter, then the internal one should be initialized.
+* @tparam FaceRange range of `boost::graph_traits<PolygonMesh>::%face_descriptor`,
+          model of `Range`. Its iterator type is `ForwardIterator`.
+* @tparam NamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+*
+* @param pmesh a polygon mesh with triangulated surface patches to be smoothed.
+* @param faces the range of triangular faces defining one or several surface patches to be smoothed.
+* @param np optional sequence of \ref pmp_namedparameters "Named Parameters" among the ones listed below.
+*
+* \cgalNamedParamsBegin
+*  \cgalParamBegin{geom_traits} a geometric traits class instance, model of `Kernel`.
+*    Exact constructions kernels are not supported by this function.
+*  \cgalParamEnd
+*  \cgalParamBegin{vertex_point_map} the property map with the points associated
+*    to the vertices of `pmesh`. Instance of a class model of `ReadWritePropertyMap`.
+*  \cgalParamEnd
+*  \cgalParamBegin{number_of_iterations} the number of iterations for the
+*    sequence of the smoothing iterations performed.
+*  \cgalParamEnd
+*  \cgalParamBegin{vertex_is_constrained_map} a property map containing the
+*    constrained-or-not status of each vertex of `pmesh`. A constrained vertex
+*    cannot be modified at all during smoothing.
+*  \cgalParamEnd
+*  \cgalParamBegin{gradient_descent_precision} precision which is achieved by minimizing the energy of each
```

Comment:
This description is a bit too long. In this package, we usually give the full description of the parameters in the "Named Parameters" page (including the default value), and a short description in the function ref manual.